openedx/core/lib/studio_entities.py

```diff
@@ -0,0 +1,64 @@
+"""Studio entity plugin manager and API."""
+import abc
+
+from openedx.core.lib.api.plugins import PluginManager
+
+
+class StudioEntityPluginManager(PluginManager):
+    """Manager for all available Studio entities.
+
+    Examples of Studio entities include Courses, Libraries, and Programs. All Studio
+    entities should implement `StudioEntity`.
+    """
+    NAMESPACE = 'openedx.studio_entity'
+
+    @classmethod
+    def get_enabled_entities(cls):
+        """Returns a list of enabled Studio entities."""
+        entities = cls.get_available_plugins()
+        enabled_entities = [entity for entity in entities.viewvalues() if entity.is_enabled()]
+
+        return enabled_entities
+
+
+class StudioEntity(object):
+    """Abstract class used to represent Studio entities.
+
+    Examples of Studio entities include Courses, Libraries, and Programs.
+    """
+    __metaclass__ = abc.ABCMeta
```

Comment:
I'm confused :-) Isn't it the case that you are both using Stevedore and requiring that each entry point subclass your abstract class. The challenge is that your repo will have to import the abstract class from edx-platform in order to extend it when it wants to provide a new extension. @cpennington, isn't that what you wanted to avoid? It will make it hard to write tests as the abstract class won't be available in your repo.